test/cea/cea708_service_unit.js

```diff
@@ -0,0 +1,677 @@
+    const text = 'test';
+    const topLevelCue = new shaka.text.Cue(startTime, endTime, '');
+    topLevelCue.nestedCues = [
+      CeaUtils.createDefaultCue(startTime, endTime, /* payload= */ text),
+    ];
+
+    const expectedCaptions = [
+      {
+        stream,
+        cue: topLevelCue,
+      },
+    ];
+
+    const captions = getCaptionsFromPackets(service, packet1, packet2);
+
+    expect(captions).toEqual(expectedCaptions);
+  });
+
+  it('handles C2 and C3 no-op control codes correctly', () => {
+    // As of CEA-708, the C2 and C3 control code group has no operations.
+    // However, the bytes are reserved for future modifications to the spec,
+    // and so the correct # of bytes should be skipped if they are seen.
+    const packets = [
+      // C2 control code data.
+      [0x1008, 0x00], // C2 Packet 1.
+      [0x1010, 0x00, 0x00], // C2 Packet 2.
+      [0x1018, 0x00, 0x00, 0x00], // C2 Packet 3.
+
+      // C3 control code data.
+      [0x1080, 0x00, 0x00, 0x00, 0x00], // C3 packet 1.
+      [0x1088, 0x00, 0x00, 0x00, 0x00, 0x00], // C3 packet 2.
+    ];
+    const expectedSkips = [1, 2, 3, 4, 5]; // As per the CEA-708-E spec.
+
+    for (let i = 0; i < packets.length; i++) {
+      const packet = createCea708PacketFromBytes(packets[i], /* pts= */ 1);
+      spyOn(packet, 'skip');
+      getCaptionsFromPackets(service, packet);
+      expect(packet.skip).toHaveBeenCalledWith(expectedSkips[i]);
+    }
+  });
+
+  describe('handles commands that change the display of windows', () => {
+    const time1 = 1;
+    const time2 = 2;
+    const time3 = 4;
+    const time4 = 5;
+    const textControlCodes = [
+      // Series of G0 control codes that add text.
+      0x74, 0x65, 0x73, 0x74, // t, e, s, t
+    ];
+
+    // These commands affect ALL windows, per the 0xff bitmap.
+    const toggleWindow = [0x8b, 0xff];
+    const displayWindow = [0x89, 0xff];
+    const deleteWindow = [0x8c, 0xff];
+    const clearWindow = [0x88, 0xff];
+
+    it('handles display, toggle, and delete commands on windows', () => {
+      // Define a visible window, add some text, and toggle it off,
+      // which should force the window to emit the caption, 'test'.
+      const packet1 = createCea708PacketFromBytes(defineWindow, time1);
+      const packet2 = createCea708PacketFromBytes(textControlCodes, time1);
+      const packet3 = createCea708PacketFromBytes(toggleWindow, time2);
+
+      // Window is now hidden. Turn it back on at time 3, and append
+      // more text to it.
+      const packet4 = createCea708PacketFromBytes(displayWindow, time3);
+      const packet5 = createCea708PacketFromBytes(textControlCodes, time3);
+
+      // Window is now being displayed. Delete all the windows.
+      // This should force the displayed window to emit the caption, 'testtest'.
+      const packet6 = createCea708PacketFromBytes(deleteWindow, time4);
+
+      // The form feed would have wiped the entire window including new
```

Comment:
The form feed control code (0x0c) is given in the control code list above, and I left a comment on it. It might just have been missed, I'll be a bit more clear in the comment.
![Screen Shot 2020-09-08 at 5 05 22 PM](https://user-images.githubusercontent.com/13709152/92527777-93f7e600-f1f5-11ea-9155-04c8509f3afa.png)

Reply:
But I don't see a form feed in **this** test case.

Reply:
You're correct.